hpx/lcos/base_lco_with_value.hpp

```diff
@@ -82,6 +83,17 @@ namespace hpx { namespace lcos
 
         virtual void set_event()
         {
+            set_event_nonvirt(std::is_default_constructible<RemoteResult>());
+        }
+
+        void set_event_nonvirt(std::false_type)
+        {
+            // this shouldn't ever be called
+            HPX_ASSERT(false);
```

Comment:
Ok, I'll change it to an exception. I was debating with myself about this but then decided to go with an assert as if this fires it wouldn't be a user error but a problem in our code.